src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -787,6 +795,15 @@ internal async Task InvokeResponse(InvocationResponse invokeResponse)
             if (_executingInvocations.TryRemove(invokeResponse.InvocationId, out ScriptInvocationContext context)
                 && invokeResponse.Result.IsInvocationSuccess(context.ResultSource, capabilityEnabled))
             {
+                var currentInvocationLatency = _workerInvocationStopWatch.GetElapsedTime();
+                _workerChannelLogger.LogDebug($"[HostMonitor] Worker {Id} invocation request took {currentInvocationLatency.TotalMilliseconds}ms");
+
+                int successfulInvocations = WorkerInvocationMetrics.IncrementSuccessfulInvocationsOfWorker(Id, _invocationMetricsPerWorkerId);
+                double averageInvocationLatency = WorkerInvocationMetrics.UpdateAverageInvocationLatency(Id, _invocationMetricsPerWorkerId, currentInvocationLatency);
+
+                string successData = $"WorkerId: {Id}, TotalInvocations : {_invocationMetricsPerWorkerId[Id].TotalInvocations}, SuccessfulInvocations : {successfulInvocations}, AverageInvocationLatency : {averageInvocationLatency}";
+                _metricsLogger.LogEvent(string.Format(MetricEventNames.WorkerInvokeSucceeded, Id), data: successData);
```

Comment:
For the initial round of changes, I was thinking we could just leverage the existing support metric events have for aggregation, and simply log count/latency events for each function completed here. We'd have event formats that include the worker Id, e.g.:

WorkerInvocationSucceeded = {0}worker.{1}.invoke.succeeded
WorkerInvocationFailed = {0}worker.{1}.invoke.failed
WorkerInvocationLatency = {0}worker.{1}.invoke.latency

For the latency event, we use BeginEvent/EndEvent as we already do for FunctionInvokeLatency. Then, MetricsEventManager will aggregate/flush these regularly. With this per worker ID data in the metrics table, we can then write Kusto queries that aggregate all this by worker ID, and we can have queries that look for apps with > 1 worker. However, I realize one problem with that is that we wouldn't be able to aggregate those events by host instance, unless we were to also include the HostInstanceId as part of the event name as well.

Using metrics events like this would give us production data we need, however I realize that if/once we decide we want to actually implement worker recycles on top of this, we'd then have to actually programmatically capture the relevant data as you're doing now.